src/main/java/org/springframework/data/tarantool/core/query/TarantoolQueryLookupStrategy.java

```diff
@@ -37,18 +37,15 @@ public RepositoryQuery resolveQuery(final Method method, final RepositoryMetadat
 
         TarantoolQueryMethod queryMethod = new TarantoolQueryMethod(method, metadata, projectionFactory);
 
-        if (hasTupleAnnotation(metadata, queryMethod)) {
-            return new TarantoolTupleRepositoryQuery(operations, queryMethod);
+        if (isOutputExpectTuple(queryMethod)) {
+            return new TarantoolRepositoryTupleQuery(operations, queryMethod);
```

Comment:
```suggestion
            return new TarantoolTupleRepositoryQuery(operations, queryMethod);
```
RepositoryQuery is a base class, be careful with changing the names since we need to keep the sources matching the Spring Data framework code.

Reply:
Okay, then I'll think about how to make the names consistent, keeping this problem in mind